### PR TITLE
Disallow special URL characters as aliases

### PR DESCRIPTION
**Pull request by arberiii**
### Description
As described on the [issue #744](https://github.com/short-d/short/issues/744), we want to disallow special characters as aliases. For now, the alias is not allowed to contain '#', but the list may be increased in the future.

### Description
Create a map with forbidden characters and this closes [issue #744](https://github.com/short-d/short/issues/744).

### CODE REVIEW

File: backend/app/usecase/validator/alias.go [hunk 1]
```diff
@@ -9,6 +9,10 @@ const (
 	customAliasMaxLength = 50
 )
 
+var forbiddenCharacters = map[rune]struct{}{
```
Comment (byliuyang) [inline backend/app/usecase/validator/alias.go:12]:
```suggestion
var forbiddenCharacters = map[rune]entity.Empty{
```

File: backend/app/usecase/validator/alias.go [hunk 2]
```diff
@@ -9,6 +9,10 @@ const (
...
+	'#': {},
```
Comment (byliuyang) [inline backend/app/usecase/validator/alias.go:13]:
```suggestion
	'#': entity.Empty{},
```

File: backend/app/usecase/validator/alias.go [hunk 3]
```diff
@@ -28,16 +32,21 @@ func (c CustomAlias) IsValid(alias *string) (bool, Violation) {
...
 		return false, HasFragmentCharacter
 	}
 
 	return true, Valid
 }
 
-// hasFragmentCharacter returns whether the alias contains the '#' character which starts fragment identifiers in URLs
-func (c CustomAlias) hasFragmentCharacter(alias string) bool {
-	return strings.ContainsRune(alias, rune('#'))
+// hasForbiddenCharacter returns whether the alias contains the one of forbidden character which starts fragment identifiers in URLs
```
Comment (byliuyang) [inline backend/app/usecase/validator/alias.go:42]:
```suggestion
// hasForbiddenCharacter returns whether the alias contains the one of forbidden character
```

File: backend/app/usecase/validator/alias.go (hunk 2 above)
Comment (arberiii) [reply backend/app/usecase/validator/alias.go:13]: This is redundant.

Comment (byliuyang) [reply backend/app/usecase/validator/alias.go:15]: @alldroll Want it to look better!

**Pull request merged**